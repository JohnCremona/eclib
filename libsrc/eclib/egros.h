// egros.h - declarations of functions for curves with good reduction outside S
//////////////////////////////////////////////////////////////////////////
//
// Copyright 2025 John Cremona
// 
// This file is part of the eclib package.
// 
// eclib is free software; you can redistribute it and/or modify it
// under the terms of the GNU General Public License as published by the
// Free Software Foundation; either version 2 of the License, or (at your
// option) any later version.
// 
// eclib is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
// for more details.
// 
// You should have received a copy of the GNU General Public License
// along with eclib; if not, write to the Free Software Foundation,
// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
// 
//////////////////////////////////////////////////////////////////////////
 

#include <set>
#include "curve.h"

// Test whether a curve with good reduction outside S and this j-invariant could exist
// (using criteria from Cremona-Lingham)
int is_j_possible(const bigrational& j, const vector<bigint>& S);

// Return integers representing QQ(S,n)
vector<bigint> twist_factors(const vector<bigint>& S, int n); // only intended for n=2,4,6

// Return list of curves with good reduction outside S and j=1728
vector<CurveRed> egros_from_j_1728(const vector<bigint>& S);
// Return list of curves with good reduction outside S and j=0
vector<CurveRed> egros_from_j_0(const vector<bigint>& S);
// Return list of curves with good reduction outside S and any fixed j
vector<CurveRed> egros_from_j(const bigrational& j, const vector<bigint>& S);

// Test whether N is a possible conductor for j=0
int is_N_possible_j_0(const bigint& N, const vector<bigint>& support);

// Test whether N is a possible conductor for j=1728
int is_N_possible_j_1728(const bigint& N, const vector<bigint>& support);
