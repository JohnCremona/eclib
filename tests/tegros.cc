// TEGROS.CC, test program for egros functions
//////////////////////////////////////////////////////////////////////////
//
// Copyright 1990-2025 John Cremona
// 
// This file is part of the eclib package.
// 
// eclib is free software; you can redistribute it and/or modify it
// under the terms of the GNU General Public License as published by the
// Free Software Foundation; either version 2 of the License, or (at your
// option) any later version.
// 
// eclib is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
// for more details.
// 
// You should have received a copy of the GNU General Public License
// along with eclib; if not, write to the Free Software Foundation,
// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
// 
//////////////////////////////////////////////////////////////////////////
//

#include <eclib/curve.h>
#include <eclib/egros.h>

int main(void)
{
  set_precision(30);
  initprimes("PRIMES",0);
  //the_primes.init(25000000);
  Curve E;

  const vector<bigint> S5 = {bigint(2), bigint(3), bigint(5)};
  const vector<bigint> S23 = {bigint(2), bigint(3)};
  const vector<bigint> S35 = {bigint(3), bigint(5)};
  const vector<bigint> S11 = {bigint(11)};

  vector<bigint> Q_S5_2 = twist_factors(S5,2);
  cout << "Q("<<S5<<", 2) = "<<Q_S5_2<<" has "<<Q_S5_2.size() << " elements (should be "<<2*pow(2,S5.size())<<")\n";
  vector<bigint> Q_S5_4 = twist_factors(S5,4);
  cout << "Q("<<S5<<", 4) has "<<Q_S5_4.size() << " elements (should be "<<2*pow(4,S5.size())<<")\n";
  vector<bigint> Q_S5_6 = twist_factors(S5,6);
  cout << "Q("<<S5<<", 6) has "<<Q_S5_6.size() << " elements (should be "<<2*pow(6,S5.size())<<")\n";
  cout<<endl;

  bigrational j(bigint(272223782641), bigint(164025));
  cout<<"Curves with good reduction outside "<<S5<<" and j = "<<j;
  if (is_j_possible(j,S5))
    cout <<" do exist"<<endl;
  else
    cout<<" do NOT exist"<<endl;
  cout<<endl;

  vector<CurveRed> egr_S23_0 = egros_from_j_0(S23);
  std::sort(egr_S23_0.begin(), egr_S23_0.end());
  cout << egr_S23_0.size()<< " curves with j=0 and good reduction outside "<<S23<<" (should be 72):\n";
  for (auto E: egr_S23_0) cout<<(Curve)E<<" conductor "<<E.conductor()<<" sort key "<<E.sort_key()<<endl;
  cout<<endl;

  vector<CurveRed> egr_S23_1728 = egros_from_j_1728(S23);
  std::sort(egr_S23_1728.begin(), egr_S23_1728.end());
  cout << egr_S23_1728.size()<< " curves with j=1728 and good reduction outside "<<S23<<" (should be 32):\n";
  for (auto E: egr_S23_1728) cout<<(Curve)E<<" conductor "<<E.conductor()<<" sort key "<<E.sort_key()<<endl;
  cout<<endl;

  cout << "Elliptic curves with conductor a power of 11, from their known j-invariants" << endl;
  vector<bigrational> j11 = {
    bigrational(bigint(-122023936), bigint(161051)),
    bigrational(bigint(-52893159101157376), bigint(11)),
    bigrational(bigint(-4096), bigint(11)),
    bigrational(bigint(-121)),
    bigrational(bigint(-32768)),
    bigrational(bigint(-24729001))
  };
  vector<CurveRed> egr_11;
  for (auto j: j11)
    {
      auto EE = egros_from_j(j, S11);
      cout << EE.size() << " curves with j = " << j << ":";
      for ( auto E: EE) cout << " " << (Curve)E;
      cout << endl;
      egr_11.insert(egr_11.end(), EE.cbegin(), EE.cend());
    }
  std::sort(egr_11.begin(), egr_11.end());
  cout<<"Sorted list:\n";
  for (auto E: egr_11)
    cout << "conductor " << E.conductor() << "\t" << (Curve)E << "\tj = " << j_invariant(E) << endl;
  cout << endl;

  vector<bigint> N_j_0, N_j_1728;
  vector<CurveRed> E_j_0, E_j_1728;
  for (int n=1; n<=100; n++)
    {
      if (!is_valid_conductor(n)) continue;
      bigint N(n);
      vector<bigint> S = pdivs(N);
      if (is_N_possible_j_0(N, S))
        {
          N_j_0.push_back(N);
          auto EE = egros_from_j_0(S);
          for (auto E: EE)
            if (E.conductor()==N)
              E_j_0.push_back(E);
        }
      if (is_N_possible_j_1728(N, S))
        {
          N_j_1728.push_back(N);
          auto EE = egros_from_j_1728(S);
          for (auto E: EE)
            if (E.conductor()==N)
              E_j_1728.push_back(E);
        }
    }
  cout << "Possible conductors <= 100 of curves with j=0:    " << N_j_0 << endl;
  cout << "Actual conductors and curves:\n";
  for (auto E: E_j_0)
    cout << E.conductor() << "\t" << (Curve)E << endl;
  cout << "Possible conductors < 100 of curves with j=1728: " << N_j_1728 << endl;
  cout << "Actual conductors and curves:\n";
  for (auto E: E_j_1728)
    cout << E.conductor() << "\t" << (Curve)E << endl;

  return 0;
}
