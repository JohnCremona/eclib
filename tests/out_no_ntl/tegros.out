Q([ 2 3 5 ], 2) = [ 1 5 3 15 2 10 6 30 -1 -5 -3 -15 -2 -10 -6 -30 ] has 16 elements (should be 16)
Q([ 2 3 5 ], 4) has 128 elements (should be 128)
Q([ 2 3 5 ], 6) has 432 elements (should be 432)

Curves with good reduction outside [ 2 3 5 ] and j = 272223782641/164025 do exist

72 curves with j=0 and good reduction outside [ 2 3 ] (should be 72):
[0,0,1,0,-7] conductor 27 sort key [ 27 0 0 0 -1 0 5 0 -7 0 0 -4 11 0 8 0 0 0 -1 5 0 -7 17 0 0 -19 0 0 1 0 -7 ]
[0,0,1,0,0] conductor 27 sort key [ 27 0 0 0 -1 0 5 0 -7 0 0 -4 11 0 8 0 0 0 -1 5 0 -7 17 0 0 -19 0 0 1 0 0 ]
[0,0,0,0,-27] conductor 36 sort key [ 36 0 0 0 -4 0 2 0 8 0 0 -4 -10 0 8 0 0 0 14 -16 0 -10 -4 0 0 14 0 0 0 0 -27 ]
[0,0,0,0,1] conductor 36 sort key [ 36 0 0 0 -4 0 2 0 8 0 0 -4 -10 0 8 0 0 0 14 -16 0 -10 -4 0 0 14 0 0 0 0 1 ]
[0,0,0,0,-108] conductor 108 sort key [ 108 0 0 0 5 0 -7 0 -1 0 0 -4 -1 0 8 0 0 0 -13 11 0 17 -13 0 0 5 0 0 0 0 -108 ]
[0,0,0,0,4] conductor 108 sort key [ 108 0 0 0 5 0 -7 0 -1 0 0 -4 -1 0 8 0 0 0 -13 11 0 17 -13 0 0 5 0 0 0 0 4 ]
[0,0,0,0,-1] conductor 144 sort key [ 144 0 0 0 4 0 2 0 -8 0 0 4 -10 0 -8 0 0 0 14 16 0 -10 4 0 0 14 0 0 0 0 -1 ]
[0,0,0,0,27] conductor 144 sort key [ 144 0 0 0 4 0 2 0 -8 0 0 4 -10 0 -8 0 0 0 14 16 0 -10 4 0 0 14 0 0 0 0 27 ]
[0,0,1,0,-1] conductor 243 sort key [ 243 0 0 0 -4 0 -7 0 -1 0 0 11 -10 0 5 0 0 0 -1 5 0 -7 -13 0 0 5 0 0 1 0 -1 ]
[0,0,1,0,20] conductor 243 sort key [ 243 0 0 0 -4 0 -7 0 -1 0 0 11 -10 0 5 0 0 0 -1 5 0 -7 -13 0 0 5 0 0 1 0 20 ]
[0,0,1,0,-61] conductor 243 sort key [ 243 0 0 0 5 0 2 0 8 0 0 -7 -1 0 -13 0 0 0 -1 5 0 -7 -4 0 0 14 0 0 1 0 -61 ]
[0,0,1,0,2] conductor 243 sort key [ 243 0 0 0 5 0 2 0 8 0 0 -7 -1 0 -13 0 0 0 -1 5 0 -7 -4 0 0 14 0 0 1 0 2 ]
[0,0,0,0,-4] conductor 432 sort key [ 432 0 0 0 -5 0 -7 0 1 0 0 4 -1 0 -8 0 0 0 -13 -11 0 17 13 0 0 5 0 0 0 0 -4 ]
[0,0,0,0,108] conductor 432 sort key [ 432 0 0 0 -5 0 -7 0 1 0 0 4 -1 0 -8 0 0 0 -13 -11 0 17 13 0 0 5 0 0 0 0 108 ]
[0,0,0,0,-16] conductor 432 sort key [ 432 0 0 0 1 0 5 0 7 0 0 4 11 0 -8 0 0 0 -1 -5 0 -7 -17 0 0 -19 0 0 0 0 -16 ]
[0,0,0,0,432] conductor 432 sort key [ 432 0 0 0 1 0 5 0 7 0 0 4 11 0 -8 0 0 0 -1 -5 0 -7 -17 0 0 -19 0 0 0 0 432 ]
[0,0,0,0,-216] conductor 576 sort key [ 576 0 0 0 -4 0 -2 0 -8 0 0 -4 10 0 -8 0 0 0 -14 16 0 -10 -4 0 0 14 0 0 0 0 -216 ]
[0,0,0,0,8] conductor 576 sort key [ 576 0 0 0 -4 0 -2 0 -8 0 0 -4 10 0 -8 0 0 0 -14 16 0 -10 -4 0 0 14 0 0 0 0 8 ]
[0,0,0,0,-8] conductor 576 sort key [ 576 0 0 0 4 0 -2 0 8 0 0 4 10 0 8 0 0 0 -14 -16 0 -10 4 0 0 14 0 0 0 0 -8 ]
[0,0,0,0,216] conductor 576 sort key [ 576 0 0 0 4 0 -2 0 8 0 0 4 10 0 8 0 0 0 -14 -16 0 -10 4 0 0 14 0 0 0 0 216 ]
[0,0,0,0,-972] conductor 972 sort key [ 972 0 0 0 -4 0 5 0 -7 0 0 -7 -10 0 -13 0 0 0 -13 11 0 17 17 0 0 -19 0 0 0 0 -972 ]
[0,0,0,0,36] conductor 972 sort key [ 972 0 0 0 -4 0 5 0 -7 0 0 -7 -10 0 -13 0 0 0 -13 11 0 17 17 0 0 -19 0 0 0 0 36 ]
[0,0,0,0,-243] conductor 972 sort key [ 972 0 0 0 -1 0 -7 0 -1 0 0 -7 11 0 -13 0 0 0 14 -16 0 -10 -13 0 0 5 0 0 0 0 -243 ]
[0,0,0,0,9] conductor 972 sort key [ 972 0 0 0 -1 0 -7 0 -1 0 0 -7 11 0 -13 0 0 0 14 -16 0 -10 -13 0 0 5 0 0 0 0 9 ]
[0,0,0,0,-12] conductor 972 sort key [ 972 0 0 0 -1 0 2 0 8 0 0 11 11 0 5 0 0 0 -13 11 0 17 -4 0 0 14 0 0 0 0 -12 ]
[0,0,0,0,324] conductor 972 sort key [ 972 0 0 0 -1 0 2 0 8 0 0 11 11 0 5 0 0 0 -13 11 0 17 -4 0 0 14 0 0 0 0 324 ]
[0,0,0,0,-3] conductor 972 sort key [ 972 0 0 0 5 0 5 0 -7 0 0 11 -1 0 5 0 0 0 14 -16 0 -10 17 0 0 -19 0 0 0 0 -3 ]
[0,0,0,0,81] conductor 972 sort key [ 972 0 0 0 5 0 5 0 -7 0 0 11 -1 0 5 0 0 0 14 -16 0 -10 17 0 0 -19 0 0 0 0 81 ]
[0,0,0,0,-32] conductor 1728 sort key [ 1728 0 0 0 -5 0 7 0 -1 0 0 4 1 0 8 0 0 0 13 11 0 17 13 0 0 5 0 0 0 0 -32 ]
[0,0,0,0,864] conductor 1728 sort key [ 1728 0 0 0 -5 0 7 0 -1 0 0 4 1 0 8 0 0 0 13 11 0 17 13 0 0 5 0 0 0 0 864 ]
[0,0,0,0,-54] conductor 1728 sort key [ 1728 0 0 0 -1 0 -5 0 7 0 0 -4 -11 0 -8 0 0 0 1 -5 0 -7 17 0 0 -19 0 0 0 0 -54 ]
[0,0,0,0,2] conductor 1728 sort key [ 1728 0 0 0 -1 0 -5 0 7 0 0 -4 -11 0 -8 0 0 0 1 -5 0 -7 17 0 0 -19 0 0 0 0 2 ]
[0,0,0,0,-2] conductor 1728 sort key [ 1728 0 0 0 1 0 -5 0 -7 0 0 4 -11 0 8 0 0 0 1 5 0 -7 -17 0 0 -19 0 0 0 0 -2 ]
[0,0,0,0,54] conductor 1728 sort key [ 1728 0 0 0 1 0 -5 0 -7 0 0 4 -11 0 8 0 0 0 1 5 0 -7 -17 0 0 -19 0 0 0 0 54 ]
[0,0,0,0,-864] conductor 1728 sort key [ 1728 0 0 0 5 0 7 0 1 0 0 -4 1 0 -8 0 0 0 13 -11 0 17 -13 0 0 5 0 0 0 0 -864 ]
[0,0,0,0,32] conductor 1728 sort key [ 1728 0 0 0 5 0 7 0 1 0 0 -4 1 0 -8 0 0 0 13 -11 0 17 -13 0 0 5 0 0 0 0 32 ]
[0,0,0,0,-144] conductor 3888 sort key [ 3888 0 0 0 -5 0 2 0 -8 0 0 7 -1 0 13 0 0 0 -1 -5 0 -7 4 0 0 14 0 0 0 0 -144 ]
[0,0,0,0,3888] conductor 3888 sort key [ 3888 0 0 0 -5 0 2 0 -8 0 0 7 -1 0 13 0 0 0 -1 -5 0 -7 4 0 0 14 0 0 0 0 3888 ]
[0,0,0,0,-81] conductor 3888 sort key [ 3888 0 0 0 -5 0 5 0 7 0 0 -11 -1 0 -5 0 0 0 14 16 0 -10 -17 0 0 -19 0 0 0 0 -81 ]
[0,0,0,0,3] conductor 3888 sort key [ 3888 0 0 0 -5 0 5 0 7 0 0 -11 -1 0 -5 0 0 0 14 16 0 -10 -17 0 0 -19 0 0 0 0 3 ]
[0,0,0,0,-9] conductor 3888 sort key [ 3888 0 0 0 1 0 -7 0 1 0 0 7 11 0 13 0 0 0 14 16 0 -10 13 0 0 5 0 0 0 0 -9 ]
[0,0,0,0,243] conductor 3888 sort key [ 3888 0 0 0 1 0 -7 0 1 0 0 7 11 0 13 0 0 0 14 16 0 -10 13 0 0 5 0 0 0 0 243 ]
[0,0,0,0,-324] conductor 3888 sort key [ 3888 0 0 0 1 0 2 0 -8 0 0 -11 11 0 -5 0 0 0 -13 -11 0 17 4 0 0 14 0 0 0 0 -324 ]
[0,0,0,0,12] conductor 3888 sort key [ 3888 0 0 0 1 0 2 0 -8 0 0 -11 11 0 -5 0 0 0 -13 -11 0 17 4 0 0 14 0 0 0 0 12 ]
[0,0,0,0,-1296] conductor 3888 sort key [ 3888 0 0 0 4 0 -7 0 1 0 0 -11 -10 0 -5 0 0 0 -1 -5 0 -7 13 0 0 5 0 0 0 0 -1296 ]
[0,0,0,0,48] conductor 3888 sort key [ 3888 0 0 0 4 0 -7 0 1 0 0 -11 -10 0 -5 0 0 0 -1 -5 0 -7 13 0 0 5 0 0 0 0 48 ]
[0,0,0,0,-36] conductor 3888 sort key [ 3888 0 0 0 4 0 5 0 7 0 0 7 -10 0 13 0 0 0 -13 -11 0 17 -17 0 0 -19 0 0 0 0 -36 ]
[0,0,0,0,972] conductor 3888 sort key [ 3888 0 0 0 4 0 5 0 7 0 0 7 -10 0 13 0 0 0 -13 -11 0 17 -17 0 0 -19 0 0 0 0 972 ]
[0,0,0,0,-648] conductor 15552 sort key [ 15552 0 0 0 -5 0 -5 0 -7 0 0 -11 1 0 5 0 0 0 -14 -16 0 -10 -17 0 0 -19 0 0 0 0 -648 ]
[0,0,0,0,24] conductor 15552 sort key [ 15552 0 0 0 -5 0 -5 0 -7 0 0 -11 1 0 5 0 0 0 -14 -16 0 -10 -17 0 0 -19 0 0 0 0 24 ]
[0,0,0,0,-18] conductor 15552 sort key [ 15552 0 0 0 -5 0 -2 0 8 0 0 7 1 0 -13 0 0 0 1 5 0 -7 4 0 0 14 0 0 0 0 -18 ]
[0,0,0,0,486] conductor 15552 sort key [ 15552 0 0 0 -5 0 -2 0 8 0 0 7 1 0 -13 0 0 0 1 5 0 -7 4 0 0 14 0 0 0 0 486 ]
[0,0,0,0,-7776] conductor 15552 sort key [ 15552 0 0 0 -4 0 -5 0 7 0 0 -7 10 0 13 0 0 0 13 -11 0 17 17 0 0 -19 0 0 0 0 -7776 ]
[0,0,0,0,288] conductor 15552 sort key [ 15552 0 0 0 -4 0 -5 0 7 0 0 -7 10 0 13 0 0 0 13 -11 0 17 17 0 0 -19 0 0 0 0 288 ]
[0,0,0,0,-6] conductor 15552 sort key [ 15552 0 0 0 -4 0 7 0 1 0 0 11 10 0 -5 0 0 0 1 -5 0 -7 -13 0 0 5 0 0 0 0 -6 ]
[0,0,0,0,162] conductor 15552 sort key [ 15552 0 0 0 -4 0 7 0 1 0 0 11 10 0 -5 0 0 0 1 -5 0 -7 -13 0 0 5 0 0 0 0 162 ]
[0,0,0,0,-96] conductor 15552 sort key [ 15552 0 0 0 -1 0 -2 0 -8 0 0 11 -11 0 -5 0 0 0 13 -11 0 17 -4 0 0 14 0 0 0 0 -96 ]
[0,0,0,0,2592] conductor 15552 sort key [ 15552 0 0 0 -1 0 -2 0 -8 0 0 11 -11 0 -5 0 0 0 13 -11 0 17 -4 0 0 14 0 0 0 0 2592 ]
[0,0,0,0,-1944] conductor 15552 sort key [ 15552 0 0 0 -1 0 7 0 1 0 0 -7 -11 0 13 0 0 0 -14 16 0 -10 -13 0 0 5 0 0 0 0 -1944 ]
[0,0,0,0,72] conductor 15552 sort key [ 15552 0 0 0 -1 0 7 0 1 0 0 -7 -11 0 13 0 0 0 -14 16 0 -10 -13 0 0 5 0 0 0 0 72 ]
[0,0,0,0,-2592] conductor 15552 sort key [ 15552 0 0 0 1 0 -2 0 8 0 0 -11 -11 0 5 0 0 0 13 11 0 17 4 0 0 14 0 0 0 0 -2592 ]
[0,0,0,0,96] conductor 15552 sort key [ 15552 0 0 0 1 0 -2 0 8 0 0 -11 -11 0 5 0 0 0 13 11 0 17 4 0 0 14 0 0 0 0 96 ]
[0,0,0,0,-72] conductor 15552 sort key [ 15552 0 0 0 1 0 7 0 -1 0 0 7 -11 0 -13 0 0 0 -14 -16 0 -10 13 0 0 5 0 0 0 0 -72 ]
[0,0,0,0,1944] conductor 15552 sort key [ 15552 0 0 0 1 0 7 0 -1 0 0 7 -11 0 -13 0 0 0 -14 -16 0 -10 13 0 0 5 0 0 0 0 1944 ]
[0,0,0,0,-288] conductor 15552 sort key [ 15552 0 0 0 4 0 -5 0 -7 0 0 7 10 0 -13 0 0 0 13 11 0 17 -17 0 0 -19 0 0 0 0 -288 ]
[0,0,0,0,7776] conductor 15552 sort key [ 15552 0 0 0 4 0 -5 0 -7 0 0 7 10 0 -13 0 0 0 13 11 0 17 -17 0 0 -19 0 0 0 0 7776 ]
[0,0,0,0,-162] conductor 15552 sort key [ 15552 0 0 0 4 0 7 0 -1 0 0 -11 10 0 5 0 0 0 1 5 0 -7 13 0 0 5 0 0 0 0 -162 ]
[0,0,0,0,6] conductor 15552 sort key [ 15552 0 0 0 4 0 7 0 -1 0 0 -11 10 0 5 0 0 0 1 5 0 -7 13 0 0 5 0 0 0 0 6 ]
[0,0,0,0,-24] conductor 15552 sort key [ 15552 0 0 0 5 0 -5 0 7 0 0 11 1 0 -5 0 0 0 -14 16 0 -10 17 0 0 -19 0 0 0 0 -24 ]
[0,0,0,0,648] conductor 15552 sort key [ 15552 0 0 0 5 0 -5 0 7 0 0 11 1 0 -5 0 0 0 -14 16 0 -10 17 0 0 -19 0 0 0 0 648 ]
[0,0,0,0,-486] conductor 15552 sort key [ 15552 0 0 0 5 0 -2 0 -8 0 0 -7 1 0 13 0 0 0 1 -5 0 -7 -4 0 0 14 0 0 0 0 -486 ]
[0,0,0,0,18] conductor 15552 sort key [ 15552 0 0 0 5 0 -2 0 -8 0 0 -7 1 0 13 0 0 0 1 -5 0 -7 -4 0 0 14 0 0 0 0 18 ]

32 curves with j=1728 and good reduction outside [ 2 3 ] (should be 32):
[0,0,0,-1,0] conductor 32 sort key [ 32 0 0 -2 0 0 6 2 0 0 -10 0 -2 10 0 0 14 0 -10 0 0 -6 0 0 10 18 0 0 0 -1 0 ]
[0,0,0,4,0] conductor 32 sort key [ 32 0 0 -2 0 0 6 2 0 0 -10 0 -2 10 0 0 14 0 -10 0 0 -6 0 0 10 18 0 0 0 4 0 ]
[0,0,0,-4,0] conductor 64 sort key [ 64 0 0 2 0 0 -6 2 0 0 10 0 2 10 0 0 -14 0 10 0 0 -6 0 0 10 18 0 0 0 -4 0 ]
[0,0,0,1,0] conductor 64 sort key [ 64 0 0 2 0 0 -6 2 0 0 10 0 2 10 0 0 -14 0 10 0 0 -6 0 0 10 18 0 0 0 1 0 ]
[0,0,0,-2,0] conductor 256 sort key [ 256 0 0 -4 0 0 -4 -2 0 0 -4 0 12 -10 0 0 -4 0 12 0 0 -6 0 0 10 -18 0 0 0 -2 0 ]
[0,0,0,8,0] conductor 256 sort key [ 256 0 0 -4 0 0 -4 -2 0 0 -4 0 12 -10 0 0 -4 0 12 0 0 -6 0 0 10 -18 0 0 0 8 0 ]
[0,0,0,-8,0] conductor 256 sort key [ 256 0 0 4 0 0 4 -2 0 0 4 0 -12 -10 0 0 4 0 -12 0 0 -6 0 0 10 -18 0 0 0 -8 0 ]
[0,0,0,2,0] conductor 256 sort key [ 256 0 0 4 0 0 4 -2 0 0 4 0 -12 -10 0 0 4 0 -12 0 0 -6 0 0 10 -18 0 0 0 2 0 ]
[0,0,0,-12,0] conductor 288 sort key [ 288 0 0 -4 0 0 -6 -8 0 0 4 0 -2 8 0 0 4 0 -10 0 0 6 0 0 -16 -18 0 0 0 -12 0 ]
[0,0,0,3,0] conductor 288 sort key [ 288 0 0 -4 0 0 -6 -8 0 0 4 0 -2 8 0 0 4 0 -10 0 0 6 0 0 -16 -18 0 0 0 3 0 ]
[0,0,0,-9,0] conductor 288 sort key [ 288 0 0 2 0 0 6 -2 0 0 10 0 -2 -10 0 0 -14 0 -10 0 0 -6 0 0 -10 18 0 0 0 -9 0 ]
[0,0,0,36,0] conductor 288 sort key [ 288 0 0 2 0 0 6 -2 0 0 10 0 -2 -10 0 0 -14 0 -10 0 0 -6 0 0 -10 18 0 0 0 36 0 ]
[0,0,0,-108,0] conductor 288 sort key [ 288 0 0 4 0 0 -6 8 0 0 -4 0 -2 -8 0 0 -4 0 -10 0 0 6 0 0 16 -18 0 0 0 -108 0 ]
[0,0,0,27,0] conductor 288 sort key [ 288 0 0 4 0 0 -6 8 0 0 -4 0 -2 -8 0 0 -4 0 -10 0 0 6 0 0 16 -18 0 0 0 27 0 ]
[0,0,0,-27,0] conductor 576 sort key [ 576 0 0 -4 0 0 6 8 0 0 4 0 2 -8 0 0 4 0 10 0 0 6 0 0 16 -18 0 0 0 -27 0 ]
[0,0,0,108,0] conductor 576 sort key [ 576 0 0 -4 0 0 6 8 0 0 4 0 2 -8 0 0 4 0 10 0 0 6 0 0 16 -18 0 0 0 108 0 ]
[0,0,0,-36,0] conductor 576 sort key [ 576 0 0 -2 0 0 -6 -2 0 0 -10 0 2 -10 0 0 14 0 10 0 0 -6 0 0 -10 18 0 0 0 -36 0 ]
[0,0,0,9,0] conductor 576 sort key [ 576 0 0 -2 0 0 -6 -2 0 0 -10 0 2 -10 0 0 14 0 10 0 0 -6 0 0 -10 18 0 0 0 9 0 ]
[0,0,0,-3,0] conductor 576 sort key [ 576 0 0 4 0 0 6 -8 0 0 -4 0 2 8 0 0 -4 0 10 0 0 6 0 0 -16 -18 0 0 0 -3 0 ]
[0,0,0,12,0] conductor 576 sort key [ 576 0 0 4 0 0 6 -8 0 0 -4 0 2 8 0 0 -4 0 10 0 0 6 0 0 -16 -18 0 0 0 12 0 ]
[0,0,0,-72,0] conductor 2304 sort key [ 2304 0 0 -4 0 0 4 2 0 0 -4 0 -12 10 0 0 -4 0 -12 0 0 -6 0 0 -10 -18 0 0 0 -72 0 ]
[0,0,0,18,0] conductor 2304 sort key [ 2304 0 0 -4 0 0 4 2 0 0 -4 0 -12 10 0 0 -4 0 -12 0 0 -6 0 0 -10 -18 0 0 0 18 0 ]
[0,0,0,-6,0] conductor 2304 sort key [ 2304 0 0 -2 0 0 -4 8 0 0 10 0 -12 -8 0 0 -14 0 -12 0 0 6 0 0 -16 18 0 0 0 -6 0 ]
[0,0,0,24,0] conductor 2304 sort key [ 2304 0 0 -2 0 0 -4 8 0 0 10 0 -12 -8 0 0 -14 0 -12 0 0 6 0 0 -16 18 0 0 0 24 0 ]
[0,0,0,-216,0] conductor 2304 sort key [ 2304 0 0 -2 0 0 4 -8 0 0 10 0 12 8 0 0 -14 0 12 0 0 6 0 0 16 18 0 0 0 -216 0 ]
[0,0,0,54,0] conductor 2304 sort key [ 2304 0 0 -2 0 0 4 -8 0 0 10 0 12 8 0 0 -14 0 12 0 0 6 0 0 16 18 0 0 0 54 0 ]
[0,0,0,-54,0] conductor 2304 sort key [ 2304 0 0 2 0 0 -4 -8 0 0 -10 0 -12 8 0 0 14 0 -12 0 0 6 0 0 16 18 0 0 0 -54 0 ]
[0,0,0,216,0] conductor 2304 sort key [ 2304 0 0 2 0 0 -4 -8 0 0 -10 0 -12 8 0 0 14 0 -12 0 0 6 0 0 16 18 0 0 0 216 0 ]
[0,0,0,-24,0] conductor 2304 sort key [ 2304 0 0 2 0 0 4 8 0 0 -10 0 12 -8 0 0 14 0 12 0 0 6 0 0 -16 18 0 0 0 -24 0 ]
[0,0,0,6,0] conductor 2304 sort key [ 2304 0 0 2 0 0 4 8 0 0 -10 0 12 -8 0 0 14 0 12 0 0 6 0 0 -16 18 0 0 0 6 0 ]
[0,0,0,-18,0] conductor 2304 sort key [ 2304 0 0 4 0 0 -4 2 0 0 4 0 12 10 0 0 4 0 12 0 0 -6 0 0 -10 -18 0 0 0 -18 0 ]
[0,0,0,72,0] conductor 2304 sort key [ 2304 0 0 4 0 0 -4 2 0 0 4 0 12 10 0 0 4 0 12 0 0 -6 0 0 -10 -18 0 0 0 72 0 ]

Elliptic curves with conductor a power of 11, from their known j-invariants
2 curves with j = -122023936/161051: [0,-1,1,-10,-20] [0,-1,1,-1250,31239]
2 curves with j = -52893159101157376/11: [0,-1,1,-7820,-263580] [0,-1,1,-946260,354609639]
2 curves with j = -4096/11: [0,-1,1,0,0] [0,-1,1,-40,-221]
2 curves with j = -121: [1,1,1,-305,7888] [1,1,0,-2,-7]
2 curves with j = -32768: [0,-1,1,-887,-10143] [0,-1,1,-7,10]
2 curves with j = -24729001: [1,1,1,-30,-76] [1,1,0,-3632,82757]

Full sorted list:
conductor 11	[0,-1,1,-7820,-263580]	j = -52893159101157376/11
conductor 11	[0,-1,1,-10,-20]	j = -122023936/161051
conductor 11	[0,-1,1,0,0]	j = -4096/11
conductor 121	[1,1,1,-305,7888]	j = -121
conductor 121	[1,1,1,-30,-76]	j = -24729001
conductor 121	[0,-1,1,-887,-10143]	j = -32768
conductor 121	[0,-1,1,-7,10]	j = -32768
conductor 121	[1,1,0,-3632,82757]	j = -24729001
conductor 121	[1,1,0,-2,-7]	j = -121
conductor 121	[0,-1,1,-946260,354609639]	j = -52893159101157376/11
conductor 121	[0,-1,1,-1250,31239]	j = -122023936/161051
conductor 121	[0,-1,1,-40,-221]	j = -4096/11

Possible conductors <= 100 of curves with j=0:    [ 27 36 72 81 ]
Actual conductors and curves:
27	[0,0,1,0,-7]
27	[0,0,1,0,0]
36	[0,0,0,0,-27]
36	[0,0,0,0,1]

Possible conductors <= 100 of curves with j=1728: [ 16 32 36 64 72 100 ]
Actual conductors and curves:
32	[0,0,0,-1,0]
32	[0,0,0,4,0]
64	[0,0,0,-4,0]
64	[0,0,0,1,0]
